Add the local settings sync blog post (#2145)
gitpod-io/website: src/routes/blog/introducing-local-settings-sync.md

---
author: filiptronicek, jeanp413, loujaybee, akosyakov
date: Monday, 23 May 2022 15:00:00 UTC
excerpt: With local Settings Sync you can sync your VS Code preferences from VS Code Browser to VS Code Desktop and vice versa.
slug: introducing-local-settings-sync
image: teaser.png
teaserImage: teaser.png

> title: Introducing VS Code local Settings Sync

Reviewer: **Suggestion:** Should this be desktop, rather than local? 🤔 

```suggestion
title: Introducing VS Code Desktop Settings Sync
```

Author: Agreed that's a more consistent alternative.

---

<script context="module">
  export const prerender = true;
</script>

> Tl;dr: from now, you can use Settings Sync with Gitpod in VS Code Desktop. You can read [the docs](https://www.gitpod.io/docs/ides-and-editors/settings-sync#enabling-settings-sync-in-vs-code-desktop) for more info.

Reviewer: **Suggestion:** Capitalise and bold "TL;DR;"

**Suggestion:** Add a note in here about the value of the feature, e.g. syncing settings browser<>desktop. 

```suggestion
**TL;DR;** from today, you can use Settings Sync with Gitpod in VS Code Desktop, allowing you to seamlessly sync your VS Code preferences from your desktop VS Code with VS Code running in the browser. You can read [the docs](https://www.gitpod.io/docs/ides-and-editors/settings-sync#enabling-settings-sync-in-vs-code-desktop) for more info.
```

Reviewer: **Suggestion:** Provide some "getting started" steps for the user to take immediately, as opposed to linking straight to the docs (which encourages the user to leave the post, and go straight to the docs). 

If you want, you could directly steal the steps from the changelog: 

https://github.com/gitpod-io/website/blob/main/src/lib/contents/changelog/2022-05-09.md

Author: @loujaybee should we then have the setups steps twice in the blog post (one at the top in the TL;DR; and the second one in the guide itself)?

I think it's a good point to not make them leave right away, but for those reading the TL;DR;, maybe they want to leave and take action ASAP? 😄

Visual Studio Code enables you to synchronize your settings, themes, extensions and other preferences across machines via a feature called [Settings Sync](https://code.visualstudio.com/docs/editor/settings-sync). This feature is convenient for both having all your machines' preferences in sync, but also for quickly bootstrapping your editor on any new machine, just the way you like it.

> Implementing Settings Sync was achieved by Gitpod making an open-source version of the Settings Sync backend, because the setting sync server that Microsoft provides with their Settings Sync service is private and therefore inaccessible to 3rd party services.

Reviewer: **Suggestion:** Do we want to mention "open-source" here? As that almost implies that the functionality is re-usable? But that's not really the/our intention? 🤔 

```suggestion
Implementing Settings Sync was achieved by creating a custom version of the Settings Sync backend, because the setting sync server that Microsoft provides with their Settings Sync service is private and therefore inaccessible to 3rd party services like Gitpod.
```

Author: I agree that your suggestion sounds better. Thanks!

> It's nice having Settings Sync work with VS Code in the browser, but how are you supposed to bring the settings you have on your local machine to Gitpod? Should you copy your whole JSON file with your settings and manually install every extension that you have locally?

Reviewer: **Suggestion:** These last two paragraphs read slightly odd for me, because we introduce the solution first, then talk about the problem. Might make sense to flip move these last two paragraphs up, so that it goes: 

1. Summary (brief problem/challenge + solution)
2. Problem / challenge
3. Solution

Author: Made something, it does not completely adhere to the structure you suggested (was harder than I thought 😄). Could you check it out, @loujaybee 🙏?
